redhat-appstudio/application-service
# [DEVHAS-109] custom kustomization patch

This PR includes the change for DEVHAS109- phase 2 in https://docs.google.com/document/d/10VEVAG2H2LpntgUV6y_ETNE1ruVuZJ2Xue1nIhlMrHI/edit#heading=h.9ck1iviree1k


With this change, custom patches will be preserved in overlays folders, and also in the kustomization file, upon any changes. 
The PR also resolves the import issue when bring user's own gitops repo, removed the `rm -rf` command when generate component to preserve user's component content

To test:
1.create `application-sample` with gitops url: https://github.com/yangcao77/petclinic-application-service-system-predict-indicate
2. create `component-sample` & `component-sample2`
3. create `snapshot` and `binding`
4. update `binding` 
5. check the overlays folder in gitops repo, the test1.yaml and test2.yaml should still persist in both overlays folder and in kustomization patches field.
